mirdata/datasets/beatles.py

```diff
@@ -141,151 +138,117 @@ def to_jams(self):
         )
 
 
-def load_audio(audio_path):
+@io.coerce_to_io(io.Mode.BYTES)
+def load_audio(faudio: BinaryIO) -> Optional[Tuple[np.ndarray, float]]:
     """Load a Beatles audio file.
 
     Args:
-        audio_path (str): path to audio file
+        faudio (str or file-like): path or file-like object pointing to an audio file
 
     Returns:
         * np.ndarray - the mono audio signal
         * float - The sample rate of the audio file
 
     """
-    if not os.path.exists(audio_path):
-        raise IOError("audio_path {} does not exist".format(audio_path))
-    return librosa.load(audio_path, sr=None, mono=True)
+    return librosa.load(faudio, sr=None, mono=True)
 
 
-def load_beats(beats_path):
+@io.coerce_to_io(io.Mode.TEXT)
+def load_beats(fbeats: TextIO) -> annotations.BeatData:
     """Load Beatles format beat data from a file
 
     Args:
-        beats_path (str): path to beat annotation file
+        fbeats (str or file-like): path or file-like object pointing to a beat annotation file
```

Comment:
same comment here - I think the (type annotations) in the docstrings can be removed if we use typing.